src/config.rs

```diff
@@ -166,6 +172,29 @@ impl OrchestratorConfig {
             warn!("no chunker configs provided");
         }
 
+        if config.passthrough_headers.is_empty() {
+            info!("No allowed headers specified");
+        }
+
+        // Add default headers to allowed_headers list
+        debug!(
+            "Adding default headers: [{}]. ",
+            DEFAULT_ALLOWED_HEADERS.join(", ")
+        );
+
+        // Lowercase all header for case-insensitive comparison
+        config.passthrough_headers = config
+            .passthrough_headers
+            .into_iter()
+            .map(|h| h.to_lowercase())
+            .collect::<HashSet<String>>();
+
+        config.passthrough_headers.extend(
+            DEFAULT_ALLOWED_HEADERS
+                .iter()
+                .map(|h| h.to_string().to_lowercase()),
```

Comment:
`String::to_lowercase()` returns a new `String`, so this performs two allocations; drop `.to_string()`.
```suggestion
                .map(|h| h.to_lowercase()),
```